contracts/StakingContract.sol

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.5;
+
+import "./Token.sol";
+
+contract StakingContract {
+  
+  Token private token;
+
+  mapping(address => uint256) public etherBalanceOf;
+  mapping(address => uint256) public depositStart;
+  mapping(address => bool) public isDeposited;
+
+  event Deposit(
+    address indexed from,
+    uint256 amount,
+    uint256 timeStart
+  );
+  event Withdraw(
+    address indexed from,
+    uint256 amount,
+    uint256 depositTime,
+    uint256 interest
+  );
+
+  constructor(Token token_) {
+    token = token_;
+  }
+
+  /**
+   * @notice Transfers an amount of ETH from the msg.sender to this contract if the sender has not 
+   * deposited before and the amount is greater than 0.01 ETH.
+   * @dev This function checks if the sender has already made a deposit, and the deposit value is
+   * greater than 0.01 ETH. If so, increases the sender's "etherBalanceOf" with the "value" being send,
+   * make the sender's status isDeposited "true" and records thh current block timestamp to a mapping.
+   * It also emits an event which has three (3) params, msg.sender, msg.value, and block.timestamp.  
+   */
+  function deposit() public payable {
+    // Checks if the user has made a deposit already
+    require(isDeposited[msg.sender] == false, "Deposit already active");
```

Comment:
is there a reason you're blocking multiple deposits? Why not allow everyone to deposit multiple times?

It would make more sense perhaps to have a maximum deposit value per user, similar to the minimum deposit you are using below. Although, in practice that's not something that is used on Staking / Pool contracts